hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionFileSystem.java

```diff
@@ -1247,4 +1149,186 @@ private static void sleepBeforeRetry(String msg, int sleepMultiplier, int baseSl
     }
     Thread.sleep((long)baseSleepBeforeRetries * sleepMultiplier);
   }
+
+  private void createWriteStrategy(Configuration conf) {
+    String className = conf.get(REGION_WRITE_STRATEGY, DefaultWriteStrategy.class.getName());
+    try {
+      LOG.info("instantiating write strategy {}", className);
+      writeStrategy = ReflectionUtils.instantiateWithCustomCtor(className,
+        new Class[] { HRegionFileSystem.class }, new Object[] { this });
+    } catch (Exception e) {
+      LOG.error("Unable to create write strategy: {}", className, e);
```

Comment:
We should bail hard if we can't load the given class -- throw it as RuntimeException

Reply:
Ack